python/ray/train/v2/api/reported_checkpoint.py

```diff
@@ -0,0 +1,19 @@
+from dataclasses import dataclass
+from typing import Any, Dict
+
+from ray.train import Checkpoint
```

Comment:
I think to avoid circular dependency we can import directly from `ray.train._checkpoint`.